eole/encoders/vision.py

```diff
@@ -88,19 +89,33 @@ class VisionEncoder(nn.Module):
     def __init__(self, encoder_config, running_config=None):
         super(VisionEncoder, self).__init__()
         self.encoder_config = encoder_config
-        self.rope = build_rope(encoder_config, mode="2d")
+        if encoder_config.position_encoding_type == PositionEncodingType.Learned:
```

Comment:
Might be worth it splitting in two separate vision encoder classes with all the conditions here. (Would probably also be more flexible to add other vision architectures down the line.)

Reply:
probably when the next encoder comes.